qrhl/qrhl.el

```diff
@@ -116,8 +129,9 @@
   proof-save-command-regexp "\\`a\\`"   ;AKA `regexp-unmatchable' in Emacs-27
   proof-tree-external-display nil
   proof-script-font-lock-keywords qrhl-font-lock-keywords
-  proof-goals-font-lock-keywords qrhl-font-lock-keywords
+  proof-goals-font-lock-keywords qrhl-font-lock-subsuperscript
   proof-response-font-lock-keywords qrhl-font-lock-keywords
+  font-lock-extra-managed-props '(display)
```

Comment:
Do you really need this? Is this variable buffer local? Otherwise you will mess with other modes.

Reply:
I definitely need it to support the `qrhl-font-lock-subsuperscript` (see above) because that one uses the `display` property to create sub-/superscripts. If we don't set `font-lock-extra-managed-props`, everything seems fine at first, but when editing `⇩x` into `⇩yx`, the `x` stays a subscript. (Because font-lock only updates the properties in `font-lock-extra-managed-props`.

But you are right, and I did not set it buffer local which needs to be fixed. I can do this by setting it in `qrhl-mode-hook` via `(set (make-local-variable 'font-lock-extra-managed-props) '(display))` instead of in `proof-easy-config`. It works but I am not sure it is the right place for this. (I am a bit unsure about when something belongs in the hook and when in the  `proof-easy-config`.) Please advise whether it should be put somewhere else.

Reply:
I think replacing `setq` by `setq-local` is enough, but I don't know how to do that with `proof-easy-config`.

Reply:
`describe-variable` on `font-lock-extra-managed-props` says that this variable becomes buffer local when set. So I guess this is OK like this.

Reply:
@Matafou In my Emacs, it does not say so! In fact, I tried it out with the current code and the effect was global. Could it be version dependent? (I used `GNU Emacs 28.1 (build 2, x86_64-w64-mingw32) of 2022-04-21`)
```
font-lock-extra-managed-props is a variable defined in ‘font-lock.el’.

Its value is (help-echo)
Local in buffer *scratch*; global value is nil

Additional text properties managed by font-lock.
This is used by ‘font-lock-default-unfontify-region’ to decide
what properties to clear before refontifying a region.

  Probably introduced at or before Emacs version 22.1.

[back]
```
I would then suggest to use `setq-local` in the hook. Does that sound right?

Reply:
I think yes, but I also think it is time to ask @monnier :-)